### DISCUSSION TITLE
I want to import a python module in my headless script? How do I do that?

### DISCUSSION THREAD

**Discussion by anzosasuke** [NationalSecurityAgency/ghidra#3577] [Q&A] [answered]
I want to import a python module in my headless script.
I tried using `import modulename` and use postscript, scriptpath and  prescript (which is obviously foolish), but I am unable to do it.
Can anyone please help me if I can import my module in another script in ghidra using headless.?

**Comment by ryanmkurtz** [comment, marked as answer]
Is the module your own code, or a built-in module?

**Reply by anzosasuke** [reply]
i want my own code.

**Reply by ryanmkurtz** [reply]
Does it work if your module lives in the same directory as the script you are running?

**Reply by anzosasuke** [reply]
If it's separately run independently run, it  runs

**Reply by anzosasuke** [reply]
my module name is xx.py and i want to import xx.py in my other script zz.py. If i run xx.py separately, it works. If i try to import xx.py, i am not able to. They both are on same directory, ghidra-scripts-master.

**Reply by ryanmkurtz** [reply]
you are just doing `import xx`?

**Reply by anzosasuke** [reply]
So is there a way to do it? if there is can you please tell me how and what should I do?

**Reply by ryanmkurtz** [reply]
I just tried to recreate your setup:

xx.py
```
print "hi"
```

zz.py
```
import xx
```
Then I ran `analyzeHeadless.bat c:\Users\xxx\ghidra_projects Test -prescript zz.py` and got
```
INFO  HEADLESS: execution starts   (HeadlessAnalyzer.java:410)
INFO  Opening existing project: c:\Users\xxx\ghidra_projects\Test   (HeadlessAnalyzer.java:1791)
INFO  Opening project: c:\Users\xxx\ghidra_projects\Test   (DefaultProject.java:134)
INFO  SCRIPT: c:\Users\xxx\ghidra_scripts\zz.py   (HeadlessAnalyzer.java:575)
hi
```
So it seems to have worked for me?  Both python files were in `~/ghidra_script` so I didn't need to specify the `-scriptPath` arg.

**Reply by anzosasuke** [reply]
oh! so if it is in ~/ghidra_scripts, it will work. thank you

**Reply by ryanmkurtz** [reply]
`~/ghidra_scripts` is on the script path by default, so you don't have to use `-scriptPath` in that case.  However, I did an extra test with putting them in a different directory and using `-scriptPath`, and my above example still worked.

**Reply by anzosasuke** [reply]
oh really! i tried using scriptPath, mine didn't work. may be i missed smthg on the commands
Thanks
